tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/PublicClientApplicationBuilderTests.cs

```diff
@@ -412,21 +412,56 @@ public void TestAuthorities()
         }
 
         [TestMethod]
-        public void TestAuthorityInvalidTenant()
+        [TestCategory("Regression")]
+        [WorkItem(1320)]
+        public void TestAuthorityWithTenant()
         {
-            try
-            {
-                var app = PublicClientApplicationBuilder.Create(TestConstants.ClientId)
-                                                    .WithAuthority(AadAuthorityAudience.AzureAdMyOrg)
-                                                    .WithTenantId("contoso.com")
-                                                    .Build();
-                Assert.Fail("Should not reach here, exception should be thrown");
-            }
-            catch (Exception ex)
-            {
-                Assert.IsTrue(ex is InvalidOperationException);
-                Assert.AreEqual(MsalErrorMessage.AzureAdMyOrgRequiresSpecifyingATenant, ex.Message);
-            }
+            var app = PublicClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                .WithAuthority(AadAuthorityAudience.AzureAdMyOrg)
+                                                .WithTenantId("contoso.com")
+                                                .Build();
+
+            Assert.AreEqual("https://login.microsoftonline.com/contoso.com/", app.Authority);
+        }
+
+        [TestMethod]
+        public void AuthorityNullArgs()
+        {
+            AssertException.Throws<ArgumentNullException>(() =>
+                PublicClientApplicationBuilder.Create(TestConstants.ClientId)
```

Comment:
is this applicable to .WithB2CAuthority?

Reply:
Sure.